### PR TITLE
Alerting: Matching instances preview for notification policies

### LINKED ISSUE #68289: Enable alerts preview on the notifications policy page
https://github.com/grafana/grafana/pull/61952 added alerts preview on the notifications policy page hidden behind a feature toggle due to performance issues.
We want to solve the issues and enable the preview without the feature toggle

### PR DESCRIPTION
**Pull request by konrad147**
**What is this feature?**
This PR adds improvements and fixes over https://github.com/grafana/grafana/pull/68291 which caused CORS violation 

The following steps have been introduced to reduce the risk associated with the feature:
- The web worker is loaded through `CorsWorker` to ensure proper functioning with CDN-hosted assets.
- Web worker loading has been wrapped in a try/catch block to handle any potential errors.
- The feature can be disabled using a feature toggle, preventing the worker code from starting the loading process.
- Components are aware that the preview engine may not be available.

Fixes: https://github.com/grafana/grafana/issues/68289

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/useRouteGroupsMatcher.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  if (routeMatcher === undefined) {
+    try {
+      worker = createWorker();
+      routeMatcher = comlink.wrap<RouteGroupsMatcher>(worker);
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        logError(e);
+      }
+      console.error('CANNOT LOAD WORKER', e);
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/useRouteGroupsMatcher.ts:31]: Do we need this if we're already calling `logError`?

Comment (konrad147) [reply public/app/features/alerting/unified/useRouteGroupsMatcher.ts:31]: No, I'll remove

**Comment by gillesdemey** [comment]
CI is failing (known issue) can you rebase from `main`?

**Review by gillesdemey — APPROVED** [review]
Awesome work! 🚀

**Review by soniaAguilarPeiron — APPROVED** [review]
awesome! 🚀

**Pull request merged**